src/patches.js

```diff
@@ -6,50 +6,92 @@ export function generatePatches(state, basePath, patches, inversePatches) {
         : generateObjectPatches(state, basePath, patches, inversePatches)
 }
 
-function generateArrayPatches(state, basePath, patches, inversePatches) {
-    const {base, copy, assigned} = state
-    const minLength = Math.min(base.length, copy.length)
+function _generateArrayPatches(
+    base,
+    copy,
+    assigned,
+    basePath,
+    patches,
+    inversePatches
+) {
+    if (copy.length < base.length) {
```

Comment:
How does this branch work? You swap `copy` with `base` and `patches` with `inversePatches`. Why?

Reply:
Swap `copy` with `base` and `patches` with `inversePatches` returns the same result, but it's normalize arrays and `delta` always be positive or 0